src/nvim/indent_c.c

```diff
@@ -508,29 +505,25 @@ get_indent_nolabel (     /* XXX */
  *   label:	if (asdf && asdfasdf)
  *		^
  */
-static int skip_label(linenr_T lnum, char_u **pp)
-{
-  char_u      *l;
-  int amount;
-  pos_T cursor_save;
+static int get_indent_after_label_pp(linenr_T lnum, char_u **pp)
+  FUNC_ATTR_NONNULL_ALL
+{ /* XXX */
+  char_u *l = ml_get(lnum);
+  int indent;
 
-  cursor_save = curwin->w_cursor;
-  curwin->w_cursor.lnum = lnum;
-  l = get_cursor_line_ptr();
-  /* XXX */
+  // We repeatedly call "ml_get(lnum)" because every time it's called (or any
+  // time we call a function that calls it, marked with an 'XXX' comment) the
+  // returned pointer is invalidated.
```

Comment:
> The arrow (^) is not aligned with the if for me, because tabs are used for indentation and their size depends on option 'tabstop'.

Ah, I see now. It wasn't aligned for me either (because I tend to use a tabstop of 2) but I didn't think to fix it. I'll fix it now.

Thanks for pointing me to the memory.c examples - I'll update the comment style.